### PR TITLE
[M] Revert "CANDLEPIN-589: Added support for forced-update owner refresh"

### PR DESCRIPTION
**Pull request by Ceiu**
This reverts commit c9ed71f4b920857564f78e6cb5eeaf8fbf404b49.

### CODE REVIEW

File: src/main/java/org/candlepin/async/tasks/RefreshPoolsJob.java [hunk 1]
```diff
@@ -178,7 +159,6 @@ public void execute(JobExecutionContext context) throws JobExecutionException {
             // Assume that we verified the request in the resource layer:
             this.refresherFactory.getRefresher(this.subAdapter, this.prodAdapter)
                 .setLazyCertificateRegeneration(lazy)
```
Comment (JoshAlbrecht23) [inline src/main/java/org/candlepin/async/tasks/RefreshPoolsJob.java:161]: Do we also want to revert the addition of `.setLazyCertificateRegeneration(lazy)`? It looks like this line was added by commit c9ed71f4b920857564f78e6cb5eeaf8fbf404b49 as well.

File: src/main/java/org/candlepin/resource/OwnerResource.java [hunk 2]
```diff
@@ -1292,8 +1292,7 @@ public AsyncJobStatusDTO refreshPools(String ownerKey, Boolean autoCreateOwner,
 
         JobConfig config = RefreshPoolsJob.createJobConfig()
             .setOwner(owner)
-            .setLazyRegeneration(true)
-            .setForceUpdate(forceUpdate != null && forceUpdate);
+            .setLazyRegeneration(true);
```
Comment (JoshAlbrecht23) [inline src/main/java/org/candlepin/resource/OwnerResource.java:1295]: Same as the other comment, just want to make sure we are intentionally keeping this new line

**Review by JoshAlbrecht23 — CHANGES_REQUESTED** [review]
I noticed a couple of lines that were left from the commit we are reverting and just want to make sure it is intentional. 

It also looks like some missing changes in src/main/java/org/candlepin/async/tasks/RefreshPoolsForProductJob.java were not reverted and I want to double check that it was intentional:
https://github.com/candlepin/candlepin/blob/c9ed71f4b920857564f78e6cb5eeaf8fbf404b49/src/main/java/org/candlepin/async/tasks/RefreshPoolsForProductJob.java#L77-L78

Also, should we add back the trace-level logging of upstream subscriptions and product data from the refresh operation?

File: src/main/java/org/candlepin/resource/OwnerResource.java (hunk 2 above)
Comment (Ceiu) [reply src/main/java/org/candlepin/resource/OwnerResource.java:1295]: The lazy regeneration wasn't added as a result of this logic, it was simply a refactor to move it out of the refresher's constructor and into more of a builder pattern. That pattern has been retained and extended slightly with Ondrej's work to break some of those circular dependencies.

**Comment by Ceiu** [comment]
The trace-level logging was a hack/workaround to get upstream data for duplicating a refresh when debugging issues in production. With the addition of direct fetching of upstream data, we no longer have a use for it here. That it was removed with the commit being reverted was happenstance due to it being in proximity of the other changes. It has no value even with the revert.

As far as the lazy regen thing goes -- that's no different than where it was originally, other than it moving to a separate method instead of being part of the refresher's constructor. The important bit is removing the forced-update param and mutator.

File: src/main/java/org/candlepin/async/tasks/RefreshPoolsJob.java (hunk 1 above)
Comment (Ceiu) [reply src/main/java/org/candlepin/async/tasks/RefreshPoolsJob.java:161]: Commented elsewhere, but this was added as part of a refactor that we're retaining despite reverting the other surrounding logic.

**Pull request merged**